woocommerce/wc-admin
# Add a country filter to the customers report

Closes #1032.

This PR adds a country filter to the customers report. Instead of making multiple requests while typing, I've added support for preloading WooCommerce's /data endpoints (they return things like countries, currencies, etc) which is useful here, and in future work like providing country drop-downs.

### Screenshots

<img width="891" alt="screen shot 2018-12-12 at 2 05 23 pm" src="https://user-images.githubusercontent.com/689165/49893352-2d277c00-fe19-11e8-8f31-b7dfc93c2fcd.png">

### Detailed test instructions:

* Open the customers report
* Add a customer filter
* Test the autocompleter, make sure countries show up correctly, and that you can select countries
* Hit filter, and make sure the query string updates
* Refresh, and make sure the previously selected countries show up
